c/zssLogging.c

```diff
@@ -43,6 +45,182 @@ bool isLogLevelValid(int level) {
   return TRUE;
 }
 
+typedef struct LogMessagePrefix_tag {
+  char text[LOG_MSG_PREFIX_SIZE];
+} LogMessagePrefix;
+
+typedef struct LogTimestamp_tag {
+  union {
+    struct {
+      char year[4];
+      char delim00[1];
+      char month[2];
+      char delim01[1];
+      char day[2];
+      char delim02[1];
+      char hour[2];
+      char delim03[1];
+      char minute[2];
+      char delim04[1];
+      char second[2];
+      char delim05[1];
+      char hundredth[2];
+    } dateAndTime;
+    char text[22];
+  };
+} LogTimestamp;
+
+
+static void getCurrentLogTimestamp(LogTimestamp *timestamp) {
+
+  uint64 stck = 0;
+  getSTCKU(&stck);
+
+  stck += getLocalTimeOffset();
+
+  stckToLogTimestamp(stck, timestamp);
```

Comment:
Is there anything better in LE? I think you can get pretty formatted timestamp using the standard library.

Reply:
It's easy to create an ISO8601 timestamp using C POSIX functions. 

```c                    
struct timeval tv;                                                  
struct tm tm;                                                       
gettimeofday(&tv, NULL);                                            
localtime_r(&tv.tv_sec, &tm);     
static const size_t ISO8601_LEN = 30;                                      
char datetime[ISO8601_LEN];                                         
char timestamp[ISO8601_LEN]; // ISO-8601 timestamp                  
strftime(datetime, sizeof datetime, "%Y-%m-%dT%H:%M:%S", &tm);      
snprintf(timestamp, sizeof timestamp, "%s.%d", datetime, tv.tv_usec)
puts(timestamp);                                                    
```